site/src/components/Dashboard/Navbar/NavbarView.tsx

```diff
@@ -214,25 +216,22 @@ const ProxyMenu: FC<ProxyMenuProps> = ({ proxyContextValue }) => {
   const isLoadingLatencies = Object.keys(latencies).length === 0;
   const isLoading = proxyContextValue.isLoading || isLoadingLatencies;
   const permissions = usePermissions();
+
   const proxyLatencyLoading = (proxy: TypesGen.Region): boolean => {
     if (!refetchDate) {
       // Only show loading if the user manually requested a refetch
       return false;
     }
 
-    const latency = latencies?.[proxy.id];
     // Only show a loading spinner if:
-    //  - A latency exists. This means the latency was fetched at some point, so the
-    //    loader *should* be resolved.
+    //  - A latency exists. This means the latency was fetched at some point, so
+    //    the loader *should* be resolved.
     //  - The proxy is healthy. If it is not, the loader might never resolve.
-    //  - The latency reported is older than the refetch date. This means the latency
-    //    is stale and we should show a loading spinner until the new latency is
-    //    fetched.
-    if (proxy.healthy && latency && latency.at < refetchDate) {
-      return true;
-    }
-
-    return false;
+    //  - The latency reported is older than the refetch date. This means the
+    //    latency is stale and we should show a loading spinner until the new
+    //    latency is fetched.
```

Comment:
Only change here is that I reformatted the comment to properly fit in an 80px column